panther_manager_bt/src/manager_node.cpp

```diff
@@ -0,0 +1,104 @@
+#include <panther_manager_bt/manager_node.hpp>
+
+namespace panther_manager_bt
+{
+
+ManagerNode::ManagerNode(ros::NodeHandle * nh)
+{
+  node_name_ = ros::this_node::getName();
+
+  const std::string default_xml =
+    ros::package::getPath("panther_manager_bt") + "/config/PantherManagerBT.xml";
+  const std::vector<std::string> default_plugin_libs = {};
+
+  xml_filename_ = nh->param<std::string>("xml_filename", default_xml);
+  plugin_libs_ = nh->param<std::vector<std::string>>("plugin_libs", default_plugin_libs);
+
+  ROS_INFO("[%s] Register BehaviorTree from: %s", node_name_.c_str(), xml_filename_.c_str());
+
+  for (const auto & p : plugin_libs_) {
+    factory_.registerFromPlugin(BT::SharedLibrary::getOSName(p));
+  }
+
+  factory_.registerBehaviorTreeFromFile(xml_filename_);
+
+  setup_behavior_tree(nh, lights_tree_, lights_config_, "Lights");
+  setup_behavior_tree(nh, safety_tree_, safety_config_, "Safety");
+  setup_behavior_tree(nh, shutdown_tree_, shutdown_config_, "Shutdown");
+
+  e_stop_sub_ = nh->subscribe("hardware/e_stop", 10, &ManagerNode::e_stop_cb, this);
+  io_state_sub_ = nh->subscribe("hardware/io_state", 10, &ManagerNode::io_state_cb, this);
```

Comment:
```suggestion
  e_stop_sub_ = nh->subscribe("hardware/e_stop", 2, &ManagerNode::e_stop_cb, this);
  io_state_sub_ = nh->subscribe("hardware/io_state", 2, &ManagerNode::io_state_cb, this);
```
Queues can be much smaller due to the nature of those topics